src/main/java/com/itrex/java/lab/repository/report/impl/ReportRepositoryImpl.java

```diff
@@ -0,0 +1,37 @@
+package com.itrex.java.lab.repository.report.impl;
+
+import com.itrex.java.lab.entity.dto.CustomerReportDTO;
+import com.itrex.java.lab.repository.report.ReportRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Repository;
+
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+import java.time.LocalDate;
+import java.util.List;
+
+@Repository
+@AllArgsConstructor
+public class ReportRepositoryImpl implements ReportRepository {
+
+    private static final String REPORT_QUERY = "select new com.itrex.java.lab.entity.dto.CustomerReportDTO" +
```

Comment:
you will have a problem if you package will be changed
create a different method to build a DTO based on the result from DB.

Reply:
It seems to me that I can't find better way for mapping resultSet with aggregation functions then I've used here (for Hibernate).
Can I use native jdbc for this purpose?